src/Sylius/Bundle/CoreBundle/EventListener/ChannelDeletionListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CoreBundle\EventListener;
+
+use Sylius\Bundle\ChannelBundle\Doctrine\ORM\ChannelRepository;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Resource\Event\ResourceEvent;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+
+/*
+ * Listener to prevent the deletion of last enabled channel.
```

Comment:
I think it's not true, listener does not prevent anything - it just listen :) What prevents it, is function inside listener. Even though, it's quite redundant and should be removed.